### PR TITLE
Add a feature to return the directories and files that are being monitored

### PR DESCRIPTION
**Pull request by NitroCao**
#### What does this pull request do?
Add a method for `Watcher` to return the directories and files that are being monitored.

#### Where should the reviewer start?
`WatchList` method in `inotify.go`.

#### How should this be manually tested?
unit testing in `inotify_test.go`.

### CODE REVIEW

**Review by PaluMacil — APPROVED** [review]
Looks appropriate and tested

**Review by nathany — CHANGES_REQUESTED** [review]
Thanks for the contribution.

Unfortunately we can't add a new `WatchList()` API that only works on Linux.

`fsnotify` is a cross-platform library. If we want to add `WatchList()`, it must also be added for macOS/BSD (kqueue) and Windows.

**Comment by mattn** [comment]
@NitroCao Would you update this?

File: inotify.go [hunk 1]
```diff
@@ -162,6 +162,20 @@ func (w *Watcher) Remove(name string) error {
 	return nil
 }
 
+// WatchList returns the directories and files that are being monitered.
+func (w *Watcher) WatchList() []string {
```
Comment (nathany) [inline inotify.go:166]: Be careful with adding new APIs that don't exist on other platforms. We want to keep a uniform API to the underlying systems.

@mattn

**Comment by chamabreu** [comment]
Hi, is this feature comming?

File: windows.go [hunk 2]
```diff
@@ -97,6 +97,19 @@ func (w *Watcher) Remove(name string) error {
 	return <-in.reply
 }
 
+// WatchList returns the directories and files that are being monitered.
+func (w *Watcher) WatchList() []string {
+	var entries []string
```
Comment (mattn) [inline windows.go:102]:
```suggestion
	entries := make([]string, len(w.watches), len(w.watches))
```

Comment (NitroCao) [reply windows.go]: Thanks for reply. updated.

**Comment by mattn** [comment]
@nathany do you have opinion others?

**Comment by nathany** [comment]
LGTM

**Comment by mattn** [comment]
Thanks. I want to confirm rules for mainteners. Only administrators have to merge this? (I know I can merge this)

image

**Comment by nathany** [comment]
Looks like there is a problem with Hound

**Comment by nathany** [comment]
I removed Hound and the branch protection rule that required it.

**Review by nathany — APPROVED** [review]

**Comment by NitroCao** [comment]
So can we merge this PR now?

**Comment by mattn** [comment]
Thank you

**Comment by nathany** [comment]
thanks @mattn 

I see there is a 1.5.3 tag that was created before the 1.5.2 tag for this. Will that cause problems getting this update?

**Comment by mattn** [comment]
OpenBSD build was broken since before. https://github.com/fsnotify/fsnotify/pull/443

**Pull request merged**